releasenotes/notes/iterable-couplingmap-b8f0cbb1b34a2005.yaml

```diff
@@ -0,0 +1,15 @@
+---
+features:
+  - |
+    :class:`.CouplingMap` is now implicitly iterable, with the iteration being
+    identical to iterating through the output of :meth:`.CouplingMap.get_edges()`.
+    In other words,
+
+    .. code-block:: python
+
+      from qiskit.transpiler import CouplingMap
+      coupling = CouplingMap.from_line(3)
+      list(coupling) == list(coupling.get_edges())
```

Comment:
You should actually be able to do

```python
      list(coupling) == coupling.get_edges()
```

Since the `EdgeList` type in rustworkx will do the equality correctly. but that doesn't matter for a release note like this.

Reply:
That's fair - it looked a bit funky to me for the list to compare equal to a "not-list" (e.g. `(1, 2, 3) != [1, 2, 3]` in Python), but if it's a defined part of the `EdgeList` struct, I guess it would have been fine.